File: .NET Word Examples/BookmarksApiDemo.cs
using Spire.Cloud.Word.Sdk.Api;
using Spire.Cloud.Word.Sdk.Client;
using System;

namespace SpireDocCloudDemo
{
    class BookmarksApiDemo
    {
        static string appId = "your id";
        static string appKey = "your key";

        public static void AddBookmark()
        {
            Configuration wordConfiguration = new Configuration(appId, appKey);
            BookmarksApi bookmarksApi = new BookmarksApi(wordConfiguration);
            string name = "addBookmark.docx";
            string startParagraphPath = "sections/0/paragraphs/0";
            int startIndexInParagraph = 0;
            string endParagraphPath = "sections/0/paragraphs/0";
            int endIndexInParagraph = 2;
            string password = null;
            string bookmarkName = "Test_bookmark";
            string folder = "input";
            string destFilePath = "output/addBookmark_output.docx";
            string storage = null;
            bookmarksApi.AddBookmark(name, startParagraphPath, startIndexInParagraph, endParagraphPath,
                endIndexInParagraph, bookmarkName, folder, storage, password, destFilePath);
        }

        public static void GetBookmarkContent()
        {
            Configuration wordConfiguration = new Configuration(appId, appKey);
            BookmarksApi bookmarksApi = new BookmarksApi(wordConfiguration);
            string name = "getBookmarkContent.docx";
            string bookmarkName = "Test";
            string folder = "input";
            string password = null;
            string storage = null;
            var response = bookmarksApi.GetBookmarkContent(name, bookmarkName, folder, storage, password);
        }
    }
}

File: Java Word Examples/BookmarksApiDemo.java
import spire.cloud.word.sdk.client.*;
import spire.cloud.word.sdk.client.api.*;
import spire.cloud.word.sdk.client.model.BookmarkInfo;

import java.util.List;

public class BookmarksApiDemo {
    static String appId = "your id";
    static String appKey = "your key";
    static String baseUrl = "https://api.e-iceblue.cn";
    static Configuration wordConfiguration = new Configuration(appId, appKey, baseUrl);
    static BookmarksApi bookmarksApi = new BookmarksApi(wordConfiguration);
    
    public static void addBookmark() throws ApiException {
        String name = "addBookmark.docx";
        String startParagraphPath = "sections/0/paragraphs/0";
        int startIndexInParagraph = 0;
        String endParagraphPath = "sections/0/paragraphs/0";
        int endIndexInParagraph = 2;
        String password = null;
        String bookmarkName = "Test_bookmark";
        String folder = "input";
        String destFilePath = "output/addBookmark_output.docx";
        String storage = null;
        bookmarksApi.addBookmark(name, startParagraphPath, startIndexInParagraph, endParagraphPath,
                endIndexInParagraph, bookmarkName, destFilePath, folder, storage, password);
    }

    public void deleteBookmark() throws Exception {
        String name = "deleteBookmark.docx";
        String storage = null;
        String folder = "Bookmark";
        String bookmarkName = "Test";
        String password = null;
        String destFilePath = "output/deleteBookmark_out.docx";
        bookmarksApi.deleteBookmark(name, bookmarkName, destFilePath, folder, storage, password);
    }


    public static void getBookmarkContent() throws ApiException {
        String name = "getBookmarkContent.docx";
        String bookmarkName = "Test";
        String folder = "input";
        String password = null;
        String storage = null;
        String response = bookmarksApi.getBookmarkContent(name, bookmarkName, folder, storage, password);
    }

    public static void getBookmarks() throws ApiException {
        String name = "getBookmarks.docx";
        String folder = "input";
        String password = null;
        String storage = null;
        List<BookmarkInfo> response = bookmarksApi.getBookmarks(name, folder, storage, password);
    }
}
